@here/harp-materials/lib/SolidLineMaterial.ts

```diff
@@ -146,20 +144,20 @@ void main() {
     float alpha = opacity;
     vec3 outputDiffuse = diffuse;
 
-    float lineEnds = max(vExtrusionCoord.x - vLength,- vExtrusionCoord.x);
-
     #if TILE_CLIP
     tileClip(vPosition.xy, tileSize);
     #endif
 
-    float pointDist = roundEdgesAndAddCaps(vSegment, vExtrusionCoord, lineEnds, vExtrusionStrength);
-    float dist = pointDist - vResultLineWidth;
+    float pointDist = roundEdgesAndAddCaps(vCoords, vRange);
+    float dist = pointDist - (lineWidth + outlineWidth) / lineWidth;
```

Comment:
what's the range of pointDist and dist? dist gets used later to modulate alpha using `(1.0 - smoothstep(-width, width, dist));` so I am assuming that should always range from `0..1` to strange artifact, but it's not clear to me why.

Reply:
- ``pointDist``: distance to line center. ``0.0`` is the line center, ``1.0`` is the line edge and any values over ``1.0`` correspond to the ``outline``.
dist: distanceToEdge. ``0.0`` marks the line+ouline edge and all negatives values correspond to pixels inside the line, while positive values correspond to pixels outside the line.

Cleaned-up and renamed this variables in latest revision to make this part of the shader code easier to understand.